### DISCUSSION TITLE
accessing tab index or name in PandasGui

### DISCUSSION THREAD

**Discussion by fdion** [adamerose/pandasgui#92] [Q&A]
Not a QT expert, so maybe this is obvious to someone out there:

Was looking at handling copy / paste based on which tab I'm on (calling grapher.copy when on grapher tab instead of dataframe_viewer.copy all the time), but I've introspected for a while this afternoon and can't figure out how to access the QTabWidget current index from PandasGui() (i.e. inside of PandasGui copy method).

Suggestions?

**Comment by adamerose** [comment]
You don't need to use the QTabWidget since I already store references to the objects in each tab on the DataFrameExplorer. I also just pushed a commit so the Plotly figure gets stored on the Grapher instead of being a temporary variable. 

```
from pandasgui.datasets import pokemon
from pandasgui import show

gui = show(pokemon)

# *Now go make a plot using Grapher*

print(gui.store.selected_pgdf.dataframe_explorer.grapher_tab)
# <pandasgui.widgets.grapher.Grapher object at 0x0000019085DA6700>

print(gui.store.selected_pgdf.dataframe_explorer.grapher_tab.figure_viewer.temp_file.name)
# C:\Users\Adam\AppData\Local\Temp\tmpiu_p49rq.html

print(gui.store.selected_pgdf.dataframe_explorer.grapher_tab.fig)

# Figure({
#     'data': [{'alignmentgroup': 'True',
#     ...
# })

```

**Reply by adamerose** [reply]
In `PandasGui` class I just have a toolbar menu option with hotkey Ctrl+C that triggers this method of itself

```
    def copy(self):
        self.store.selected_pgdf.dataframe_viewer.copy()
```

**Reply by fdion** [reply]
Yes, that's why I was asking how I can figure which tab is active. dataframe_viewer.copy() shouldn't get called when doing ctrl-c on something on the grapher tab, for example. If you try, you'll get an error... should just go to clipboard in those cases.

**Reply by adamerose** [reply]
> Yes, that's why I was asking how I can figure which tab is active. dataframe_viewer.copy() shouldn't get called when doing ctrl-c on something on the grapher tab, for example. If you try, you'll get an error... should just go to clipboard in those cases.

Oh now I get what you mean... well this is actually a bit tricky because it uses QDockWidget instead of QTabbedWidget, so it's hard to determine which tab is active. For example you might have a split view like this
![image](https://user-images.githubusercontent.com/10884874/106979818-16e18600-672d-11eb-9e40-84b5a29bae54.png)

Your question also made me realize that by moving the copy/paste hotkeys to the top level PandasGui menu I made users unable to copy cells from the Statistics tab. I need to rethink this... I wanted Copy and Copy With Header to be in the menu bar so users were aware of the hotkeys but may need to undo that.


I'm working on a way to detect the last clicked dock, I'll probably push it tonight

**Reply by adamerose** [reply]
BTW I think for anything specific like this it makes sense to use GitHub issues so I can reference it in commits and they are usually closeable when implemented/fixed. Discussion seems better suited for broad topics, usage Q/A or demos

**Reply by fdion** [reply]
Yeah, I guess that makes sense. Just trying to figure out the best way to brainstorm, hence more discussion type stuff.
